app/src/main/java/com/parking_system_kotlin/mvp/view/base/ActivityView.kt

```diff
@@ -0,0 +1,25 @@
+package com.parkingsystemkotlin.mvp.view.base
+
+import android.app.Activity
+import android.content.Context
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.FragmentManager
+
+import java.lang.ref.WeakReference
+
+open class ActivityView(activity: Activity) {
+    private var activityRef: WeakReference<Activity> = WeakReference(activity)
+
+    fun getActivity(): Activity? {
+        return activityRef.get()
+    }
```

Comment:
here you can change this method to a variable. and re-define the getter 
See the counter project eaxmple

Reply:
done